File: server/router/system/sender_email_account.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type SenderEmailAccountRouter struct{}

func (r *SenderEmailAccountRouter) InitSenderEmailAccountRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	group := Router.Group("senderEmailAccount").Use(middleware.OperationRecord())
	groupWithoutRecord := Router.Group("senderEmailAccount")
	{
		group.POST("create", senderEmailAccountApi.CreateSenderEmailAccount)
		group.PUT("update", senderEmailAccountApi.UpdateSenderEmailAccount)
		group.PUT("updateQuota", senderEmailAccountApi.UpdateSenderEmailAccountQuota)
		group.DELETE("delete", senderEmailAccountApi.DeleteSenderEmailAccount)
		group.POST("setDefault", senderEmailAccountApi.SetDefaultSenderEmailAccount)
		group.POST("testById", senderEmailAccountApi.TestSenderEmailAccountConnectionByID)
	}
	{
		groupWithoutRecord.GET("list", senderEmailAccountApi.GetSenderEmailAccountList)
		groupWithoutRecord.POST("test", senderEmailAccountApi.TestSenderEmailAccountConnection)
	}
}

File: web/src/api/senderEmailAccount.js
import service from '@/utils/request'

export const getSenderEmailAccountList = () => {
  return service({
    url: '/senderEmailAccount/list',
    method: 'get'
  })
}

export const createSenderEmailAccount = (data) => {
  return service({
    url: '/senderEmailAccount/create',
    method: 'post',
    data
  })
}

export const updateSenderEmailAccount = (data) => {
  return service({
    url: '/senderEmailAccount/update',
    method: 'put',
    data
  })
}

export const updateSenderEmailAccountQuota = (data) => {
  return service({
    url: '/senderEmailAccount/updateQuota',
    method: 'put',
    data
  })
}

export const deleteSenderEmailAccount = (data) => {
  return service({
    url: '/senderEmailAccount/delete',
    method: 'delete',
    data
  })
}

export const setDefaultSenderEmailAccount = (data) => {
  return service({
    url: '/senderEmailAccount/setDefault',
    method: 'post',
    data
  })
}

export const testSenderEmailAccount = (data) => {
  return service({
    url: '/senderEmailAccount/test',
    method: 'post',
    data
  })
}

export const testSenderEmailAccountById = (data) => {
  return service({
    url: '/senderEmailAccount/testById',
    method: 'post',
    data
  })
}

